### ISSUE TITLE
[Suggestion] Syllable types -> Words structure

### ISSUE THREAD

**Issue by ItzShiney** [DraqueT/PolyGlot#1173]
Some languages can have words with no consonant clusters and with. It's not very hard to implement such a language now, but it requires some effort. The suggestion is to allow regex-like notation in syllable types field and rename it words structure. Eg. all words in a conlang:

- have structure `(C)VCV(C)` => write `C?VCVC?` instead of `VCV CVCV VCVC CVCVC`
- consist of 1 to 4 syllables which have structure `(C)V(N)` => write `(C?VN?){1:4}` instead of `V CV VN CVN VV CVV CVC` and so on
- (the case which made me write this issue) can have up to one aspirated vowel and up to one long vowel, and if both are present then they go one after another & have structure `(C)VCV(C)` => write `[AN]?[VL]NVF? N?V[AN][VL]F?` (where N is a non-aspirated consonant, A is an aspirated one, V is a short vowel and L is a long vowel)

Maybe the functionality can be imitated with rewrite rules & illegal clusters, but it is obviously a bad approach

**Comment by DraqueT** [comment]
For clarification, is this in the word generation section?

**Comment by ItzShiney** [comment]
It is

**Comment by DraqueT** [comment]
I am going to have to give this some thought. This module is a direct adaptation of the Zompist word generator, and a lot of users have expressed that they want to be able to copy/paste the patterns they use on that site into PolyGlot. This is absolutely one that I'm going to let stew in my brain for a bit though, it is a good idea, and if I'm careful about implementing it, I think I could add syntax that leaves Zompist functionality intact.

**Comment by ItzShiney** [comment]
Yeah, maybe two different tabs? One is current `Word Generator`, left unchanged, and one is `Advanced Word Generator` working like I described. The tabs share the `Categories`, `Illegal Clusters` and `Rewrite Rules` fields, but `Advanced Word Generator` has `Word Types` field instead of `Syllable Types`, and has none of these↓ settings
![image](https://user-images.githubusercontent.com/24991493/169396654-2ded1df0-4f80-46ec-90d1-ab9807693d5b.png)
